### DISCUSSION TITLE
Stuck on 16.4

### DISCUSSION THREAD

**Discussion by awptechnologies** [docker-library/postgres#1285] [General]
I am stuck on the 16.4 image. If i go back to latest it starts and immediately crashes. The latest image i can run is 16.4. Is there anything i can do to fix this?

**Comment by yosifkit** [comment]
I think you'd need to provide more information in order for someone to reproduce the failure and be able to offer a solution.

If I had a guess, it would be that it is related to the major version upgrade. Did you follow the [PostgreSQL upgrading guide](https://www.postgresql.org/docs/17/upgrading.html) to move major versions (16 -> 17). There are also proof of concept examples (https://github.com/tianon/docker-postgres-upgrade) and projects (https://github.com/pgautoupgrade/docker-pgautoupgrade) that can help.

**Reply by awptechnologies** [reply]
I see the upgrade guide. Do i need to follow this even if it is a docker image?

**Reply by tianon** [reply]
This Docker image does not provide any automatic functionality related to PostgreSQL major version upgrades (because they are quite complicated and require _both_ the old version _and_ the new version installed concurrently or a full database dump/reimport to accomplish successfully).

**Reply by awptechnologies** [reply]
So any recommendations for me to get to latest without starting over?

**Reply by tianon** [reply]
> There are also proof of concept examples (https://github.com/tianon/docker-postgres-upgrade) and projects (https://github.com/pgautoupgrade/docker-pgautoupgrade) that can help.

(from the message above)

See also #37

In the interim (and long-term, even), you should probably pin to a specific major version (`postgres:16` instead of `postgres:latest`, for example).
